src/async-client/index.ts

```diff
@@ -49,29 +49,27 @@ export const asyncClient = <S extends Loadable<unknown>>(
       }
       return storeValue;
     },
+    set: (proxiedFunction, property, value) => {
+      return Reflect.set(loadable, property, value);
+    },
+    defineProperty(proxiedFunction, property, value) {
+      return Reflect.defineProperty(loadable, property, value);
+    },
```

Comment:
Previously, if one mocked the implementation of .load() on an asyncClient, it would never resolve because [the getter always pointed to the target client](https://github.com/square/svelte-store/blob/main/src/async-client/index.ts#L36).